src/app/layout/Breadcrumb.tsx

```diff
@@ -1,44 +1,55 @@
 import React from 'react'
-import { Link, useMatches } from 'react-router-dom'
-import {
-    Section,
-    isOverviewSection,
-    getSectionPath,
-    getOverviewPath,
-    useToWithSearchState,
-} from '../../lib'
+import { Link, To, useLocation, useMatches, matchPath } from 'react-router-dom'
+import { useToWithSearchState } from '../../lib'
 import type { MatchRouteHandle } from '../routes/types'
 import css from './Breadcrumb.module.css'
 
 const BreadcrumbSeparator = () => <span className={css.separator}>/</span>
 
 type BreadcrumbItemProps = {
-    section: Section
-    label?: string
+    label: string
+    to: To
 }
 
-export const BreadcrumbItem = ({ section, label }: BreadcrumbItemProps) => {
-    const isOverview = isOverviewSection(section)
-    const link = isOverview ? getOverviewPath(section) : getSectionPath(section)
-    const to = useToWithSearchState(`/${link}`)
+export const BreadcrumbItem = ({ label, to }: BreadcrumbItemProps) => {
+    const resolvedTo = useToWithSearchState(to)
```

Comment:
super small, but can we rename this to something that makes more sense. I had to go read the function to understand what this does. Something like href?

Reply:
Do you have a suggestion? Naming these `searchState` things was kind of hard. Realizing that `to` is a `react-router` concept/type is kind of necessary for the name to make sense, so I agree it could be better. 

- `useApplySearchState(to: To)`
- `useToWithPreservedSearchState(to: To)`